runtime/shared_common/OSCache.cpp

```diff
@@ -107,8 +108,19 @@ SH_OSCache::getCacheVersionAndGen(J9PortLibrary* portlib, J9JavaVM* vm, char* bu
 	} else if (J9PORT_SHR_CACHE_TYPE_SNAPSHOT == versionData->cacheType) {
 		versionStr[versionStrLen] = J9SH_SNAPSHOT_PREFIX_CHAR;
 	}
+	if (generation < J9SH_GENERATION_37) {
+		j9str_printf(PORTLIB, genString, 4, "G%02d", generation);
+	} else if (J9SH_GENERATION_37 == generation) {
+		/* For generation 37, it is possible that there is no layer number in the cache file name */
```

Comment:
If it makes the code simpler, the generation number can be changed to 39 concurrently with adding the layer number.

Reply:
Increased the generation number to 39.